includes/utils/class-amp-validation-utils.php

```diff
@@ -231,14 +301,133 @@ public static function validate_content( $post ) {
 		}
 	}
 
+	/**
+	 * Wraps callbacks in comments, to indicate to the sanitizer which plugin added them.
+	 *
+	 * Iterates through all of the registered callbacks.
+	 * If a callback is from a plugin and outputs markup,
+	 * this wraps the markup in comments.
+	 * Later, the sanitizer can identify which plugin any illegal markup is from.
+	 *
+	 * @global array $wp_filter
+	 * @return void
+	 */
+	public static function callback_wrappers() {
+		global $wp_filter;
+		if ( ! self::do_validate_front_end() ) {
+			return;
+		}
+		$pending_wrap_callbacks = array();
+		foreach ( $wp_filter as $filter_tag => $wp_hook ) {
+			foreach ( $wp_hook->callbacks as $priority => $callbacks ) {
+				foreach ( $callbacks as $callback ) {
+					$plugin = self::get_plugin( $callback['function'] );
```

Comment:
Also needs to account for `themes` and `mu-plugins`.

Reply:
This now accommodates themes and plugins, via the new function [get_source()](https://github.com/Automattic/amp-wp/pull/971/files#diff-7d234989187d87a372e2aa6aa1299a71R349).